multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ElementUpdater.java

```diff
@@ -30,7 +39,14 @@
 
         @Override
         public <T> List<T> updateList(List<T> oldList, List<T> newList) {
-            return ListUtils.union(oldList, newList);
+            List<T> newPartsOnly = ListUtils.subtract(newList, oldList);
+            return ListUtils.union(oldList, newPartsOnly);
```

Comment:
Previously this will result in List with duplicated old elements in result list, is it right?

Reply:
yes, this was a defect previously; thankfully we also didn't have a real case which would've used this code